roles/container-host/tasks/container_mirror.yml

```
@@ -0,0 +1,37 @@
+---
+- name: "Create {{ container_mirror_cert_path }}"
+  file:
+    path: "{{ container_mirror_cert_path }}"
+    state: directory
+
+- name: "Copy {{ container_mirror }} self-signed cert"
+  copy:
+    dest: "{{ container_mirror_cert_path }}/docker-mirror.crt"
+    content: "{{ container_mirror_cert }}"
+
+- name: Install registries-conf-ctl 
+  pip:
+    name: git+https://github.com/sebastian-philipp/registries-conf-ctl
+    state: latest
+
+# Why is this even necessary?  I couldn't figure this out.  I'd pip install but the command was not found in the next task.  Tried '--user', umask: 0022, shell and command modules.
+- name: Find registries-conf-ctl
+  stat:
+    path: /usr/bin/registries-conf-ctl
+  register: usr_bin_rcc
+
+- name: Find registries-conf-ctl again
+  stat:
+    path: /usr/local/bin/registries-conf-ctl
+  register: usr_local_bin_rcc
+
+- set_fact:
+    rcc_path: /usr/bin/registries-conf-ctl
+  when: usr_bin_rcc.stat.exists
+
+- set_fact:
+    rcc_path: /usr/local/bin/registries-conf-ctl
+  when: usr_local_bin_rcc.stat.exists
+
+- name: Add local docker.io registry mirror
+  command: "{{ rcc_path }} --conf={{ container_service_conf }} add-mirror docker.io {{ container_mirror }}"
```

```suggestion
  command: "{{ rcc_path }} add-mirror docker.io {{ container_mirror }}"
```

There should not be a need to specify the config anymore.

---

If both exist, will they get modified?

---

yes!

---

Something's still off then.  This was on an Ubuntu 16.04 machine with docker.io installed.  It shouldn't be looking for `/etc/containers/registries.conf`

```
TASK [container-host : Add local docker.io registry mirror] ********************************************************************************************************************************************************************************************************************
task path: /home/dgalloway/git/ceph/ceph-cm-ansible/roles/container-host/tasks/container_mirror.yml:53
changed: [gibba036.front.sepia.ceph.com] => {
    "changed": true, 
    "cmd": [
        "/usr/local/bin/registries-conf-ctl", 
        "add-mirror", 
        "docker.io", 
        "docker-mirror.front.sepia.ceph.com:5000"
    ], 
    "delta": "0:00:00.069744", 
    "end": "2020-12-03 18:55:43.784405", 
    "invocation": {
        "module_args": {
            "_raw_params": "/usr/local/bin/registries-conf-ctl add-mirror docker.io docker-mirror.front.sepia.ceph.com:5000", 
            "_uses_shell": false, 
            "argv": null, 
            "chdir": null, 
            "creates": null, 
            "executable": null, 
            "removes": null, 
            "stdin": null, 
            "stdin_add_newline": true, 
            "strip_empty_ends": true, 
            "warn": true
        }
    }, 
    "rc": 0, 
    "start": "2020-12-03 18:55:43.714661", 
    "stderr": "", 
    "stderr_lines": [], 
    "stdout": "Ex: [Errno 2] No such file or directory: '/etc/containers/registries.conf'", 
    "stdout_lines": [
        "Ex: [Errno 2] No such file or directory: '/etc/containers/registries.conf'"
    ]
}
```

---

:eyes:

---

`"rc": 0, `

---

fixed in https://github.com/sebastian-philipp/registries-conf-ctl/commit/0aca9c6f4412cc8dc0177c60176980148b8ad3d9